x-pack/platform/plugins/shared/entity_manager/server/lib/v2/queries/entity_count.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { compact, last } from 'lodash';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import { EntitySourceDefinition } from '../types';
+
+const fromCommand = ({ sources }: { sources: EntitySourceDefinition[] }) => {
+  let command = `FROM ${sources.flatMap((source) => source.index_patterns).join(', ')}`;
+  if (sources.length > 1) {
+    command += ' METADATA _index';
+  }
+
+  return command;
+};
+
+const dslFilter = ({
+  sources,
+  filters,
+  start,
+  end,
+}: {
+  sources: EntitySourceDefinition[];
+  filters: string[];
+  start: string;
+  end: string;
+}) => {
+  const sourcesFilters = sources
+    .map((source) => {
+      const sourceFilters = [
+        ...source.filters,
+        ...source.identity_fields.map((field) => `${field}: *`),
+      ];
+
+      if (source.timestamp_field) {
+        sourceFilters.push(
+          `${source.timestamp_field} >= "${start}" AND ${source.timestamp_field} <= "${end}"`
+        );
+      }
+
+      sourceFilters.push(
+        source.index_patterns
+          .map((pattern) => `_index: "${pattern}*" OR _index: ".ds-${last(pattern.split(':'))}*"`)
+          .join(' OR ')
+      );
```

Comment:
Why is this needed?

Reply:
contrived example but take these two sources sharing the same type
- `{ patterns: "foo*", identity: "service.name", filters: "service.name: foo" }`
- `{ patterns: "bar*", identity: "service.name", filters: "service.name: bar" }`

now say foo* is empty and bar* contains `{ service.name: "foo" }` and `{ service.name: "bar" }`. without the index filter we'd count a total of 2 services when according to the sources it should be 1